src/team_lib/ucx/xccl_ucx_lib.c

```diff
@@ -123,11 +129,39 @@ xccl_ucx_allreduce_init(xccl_coll_op_args_t *coll_args,
                         xccl_tl_coll_req_t **request, xccl_tl_team_t *team)
 {
     //TODO alg selection for allreduce should happen here
-    xccl_ucx_collreq_t *req;
+    xccl_ucx_collreq_t          *req;
+    xccl_status_t               status;
+    xccl_team_lib_ucx_context_t *ctx;
+    int                         alg_id;
+     
     xccl_ucx_coll_base_init(coll_args, team, &req);
-    req->start = xccl_ucx_allreduce_knomial_start;
-    (*request) = (xccl_tl_coll_req_t*)&req->super;
-    return XCCL_OK;
+    ctx = ucs_derived_of(team->ctx, xccl_team_lib_ucx_context_t); 
+    
+    if (coll_args->alg.set_by_user) {
+        alg_id = coll_args->alg.id;
+    } else {
+        alg_id = ctx->allreduce_alg_id;
```

Comment:
i would probably also added smth like: default value for allreduce_alg_id = -1 meaning "auto". For msgsize < e.g. 4K use knomial small, otherwise use RSA

Reply:
replaced alg_id with alg names and added auto